app.rb

```diff
@@ -0,0 +1,87 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield self[i], i
+      i += 1
+    end
+  end
+
+  def my_select
+    new_numbers = []
+    each do |i|
+      new_numbers.push(i) if yield(i)
+    end
+    new_numbers
+  end
+
+  def my_all?
+    condition = true
+    each do |j|
+      condition = false unless yield(j)
+    end
+    condition
+  end
+
+  def my_any?
+    each do |j|
+      return true if yield(j)
+    end
+    false
+  end
```

Comment:
1. When no block or argument is given should return true if at least one of the collection is not false or nil or false if at least one of the collection is not true. For example:
`[1, 'dog' , 1].my_any?` - should return true

2. When a class is passed as an argument should return true if at least one of the collection is a member of such class::Integer. For example:
`['a', 'b', 3].my_any?(Integer)` - should return true

3. When a class is passed as an argument should return true if at least one of the collection is a member of such class::Numeric. For example:
`['b', 2i, 'a'].my_any?(Numeric)` - should return true

4. When a Regex is passed as an argument returns true if any of the collection matches the Regex or false if none of the collection matches the Regex. For example:
`['dog','door','dish'].my_any?(/s/)` - should return true

5. When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern or
true if any of the collection matches the pattern. For example:
`['dog','door','dish'].my_any?(/dog/)` - should return true